proxy/envoy/config.go

```diff
@@ -126,9 +126,47 @@ func Generate(instances []*model.ServiceInstance, services []*model.Service, rul
 // buildListeners uses iptables port redirect to route traffic either into the
 // pod or outside the pod to service clusters based on the traffic metadata.
 func buildListeners(instances []*model.ServiceInstance,
-	services []*model.Service,
+	services []*model.Service, rules []*config.RouteRule,
+	upstreams []*config.UpstreamCluster,
```

Comment:
Can we not use the word upstream here? I find "upstream cluster" confusing since we are dealing with many proxies in many roles and also many actual clusters for future federated solution. Inbound - outbound makes more sense in this context, or maybe use or introduce terminology to the model